### ISSUE TITLE
v1.13 and v1.14 have issues building

### ISSUE THREAD

**Issue by alexcouper** [ncw/rclone#73]
Very likely that I'm not doing something right.

I can build v1.12 fine - but v1.13 and v1.14 fail.

```
$ git checkout v1.12
Previous HEAD position was 88293a4... Version v1.13
HEAD is now at ba20917... Version v1.12
$ go build

$ git checkout v1.13
Previous HEAD position was ba20917... Version v1.12
HEAD is now at 88293a4... Version v1.13
$ go build
# github.com/ncw/rclone/s3
s3/s3.go:187: c.Client undefined (type *s3.S3 has no field or method Client)

$ git checkout v1.14
Previous HEAD position was 88293a4... Version v1.13
HEAD is now at d9fcc32... Version v1.14
$ go build
# github.com/ncw/rclone/s3
s3/s3.go:187: c.Client undefined (type *s3.S3 has no field or method Client)
# github.com/ncw/rclone/googlecloudstorage
googlecloudstorage/googlecloudstorage.go:45: undefined: storage.DevstorageFullControlScope
```

**Comment by ncw** [comment]
rclone had quite a few dependencies and I don't (yet) version these alongside rclone, so if the dependencies change then you'll see errors like the above.

It isn't entirely satisfactory I know, but I haven't felt like doing the whole vendoring thing until the community works out which direction it is going it!

[Here is part of the discussion](https://groups.google.com/forum/#!msg/golang-dev/nMWoEAG55v8/iJGgur7W_SEJ)

So in summary, the above is unfortunate but expected.

**Comment by zeshanb** [comment]
Most likely cause could be the generation of client id and secret in Google
cloud console.

I would login to google cloud console and generate a client id and secret.

https://console.developers.google.com

After selecting your project -> Go to APIs & Auth from left menu.

Create oAuth id

Select "installed Application" that is Other

This will display the client id and secret.

All the best..

Regards,
Zeshan

**Comment by ncw** [comment]
@zeshanb it is to do with the fact that google changed the API of their go cloudstorage module recently rather than any config change

**Comment by zeshanb** [comment]
@nick oh okay. thanks.

**Comment by ncw** [comment]
`go get -u -v -f` to update the stale dependencies should fix the build for the latest checkout `git checkout master`.

The older versions don't build because the libraries they depend on have changed.

**Comment by alexcouper** [comment]
I realize you aren't "Go support" and my issues seem to be a more general "how do you upgrade dependencies", so whilst I appreciate any help - I also completely understand if you want to say "I think you should spend some time reading the go tutorials".

```
$ go get -u -v -f
flag provided but not defined: -f
usage: get [-d] [-fix] [-t] [-u] [build flags] [packages]
```

```
$ go get -fix -u -v
github.com/ncw/rclone (download)
code.google.com/p/goauth2 (download)
github.com/Unknwon/goconfig (download)
github.com/mreiferson/go-httpclient (download)
github.com/ncw/goamz (download)
github.com/ogier/pflag (download)
github.com/tsenart/tb (download)
golang.org/x/net (download)
google.golang.org/api (download)
github.com/stacktic/dropbox (download)
golang.org/x/oauth2 (download)
github.com/ncw/swift (download)
github.com/ncw/rclone/googlecloudstorage
github.com/ncw/rclone/s3
# github.com/ncw/rclone/googlecloudstorage
googlecloudstorage/googlecloudstorage.go:45: undefined: storage.DevstorageFullControlScope
# github.com/ncw/rclone/s3
s3/s3.go:187: c.Client undefined (type *s3.S3 has no field or method Client)
```

**Comment by ncw** [comment]
I should read my own docs - see the RELEASE.md file!  It should be `go get -u -f -v ./...`

If you haven't got `-f` then you are using go 1.3, so leave it out `go get -u -v ./...` (or upgrade to go 1.4)

If you really can't get the upgrading to work, then delete `$GOPATH/src/google.golang.org/api` and then `go get google.golang.org/api/...`

Hope that helps!
